### PR TITLE
Adding exception types

### PR DESCRIPTION
**Pull request by coryodaniel**
Adding types for exceptions so consuming libraries can reference them in typespecs.

### CODE REVIEW

File: lib/yaml_elixir/exceptions.ex [hunk 1]
```diff
@@ -1,9 +1,16 @@
 defmodule YamlElixir.FileNotFoundError do
   defexception message: "file error"
+  @type t :: %__MODULE__{message: binary}
```
Comment (KamilLelonek) [inline lib/yaml_elixir/exceptions.ex]: It's not just a binary. Please use a proper String type instead.

File: lib/yaml_elixir/exceptions.ex [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 end
 
 defmodule YamlElixir.ParsingError do
   defexception [:line, :column, :type, message: "parsing error"]
+  @type t :: %__MODULE__{
+    line: integer | nil,
+    column: integer | nil,
```
Comment (KamilLelonek) [inline lib/yaml_elixir/exceptions.ex]: Please use parenthesis for the types.

File: lib/yaml_elixir/exceptions.ex [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
+    type: term | nil,
```
Comment (KamilLelonek) [inline lib/yaml_elixir/exceptions.ex]: Are you sure `type` can be `any()`?

**Review by KamilLelonek — CHANGES_REQUESTED** [review]

File: lib/yaml_elixir/exceptions.ex (hunk 3 above)
Comment (coryodaniel) [reply lib/yaml_elixir/exceptions.ex]: Yikes, that was supposed to be atom.

**Review by KamilLelonek — APPROVED** [review]

**Comment by coryodaniel** [comment]
Is there anything else I need to do to get this released?

**Comment by KamilLelonek** [comment]
Merged & published.

**Comment by coryodaniel** [comment]
Awesome, thank you!

**Pull request merged**